### DISCUSSION TITLE
MySQL introspection: TINYINT is typed as generated<number> - is this by design?

### DISCUSSION THREAD

**Discussion by cheekybuddha** [RobinBlomberg/kysely-codegen#205] [Q&A]
Hello,

I just wanted to check whether this might be a bug or, more likely, if I'm just missing something!

With a table like:
``` sql
CREATE TABLE `course_levels` (
  `id` int NOT NULL AUTO_INCREMENT,
  `level` varchar(50) NOT NULL,
  `sort_order` tinyint DEFAULT '1',
  `valid_from` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `valid_until` datetime DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `uidx_level` (`level`),
  KEY `idx_valid_from` (`valid_from`),
  KEY `idx_valid_until` (`valid_until`)
) ENGINE=InnoDB AUTO_INCREMENT=8 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
```
the generated type comes out as:
``` ts
export interface CourseLevels {
  id: Generated<number>;
  level: string;
  sort_order: Generated<number | null>;
  valid_from: Generated<Date>;
  valid_until: Date | null;
}
```
which means I can't insert/update values for sort_order field.

I use the following command:
``` bash
npx kysely-codegen --out-file /home/me/proj1/src/lib/server/db/db.t.ts
```

Why is it typed as ``` Generated<number | null> ``` instead of ``` number | null ``` ?

Thanks in advance, and thank you for this amazing tool!

[Edit]
Using:
kysely 0.27.4
kysely-codegen 0.16.8

**Comment by cheekybuddha** [comment]
... or is it Generated because there is a default value?

**Comment by RobinBlomberg** [comment]
Hello, and thanks for putting this in the Q&A!

That column is marked as `Generated` because it has a default database value, which means that you don't _need_ to supply a value when inserting or updating. However, you are in fact completely free to insert/update with your own value. 👍

**Comment by cheekybuddha** [comment]
Thanks for the response! 👍

I realised I was barking up the wrong tree as I try to track down why something isn't working as expected :oops: 😬

Typescript can be confusing and kysely-codegen is a lifesaver!

**Comment by RobinBlomberg** [comment]
Appreciated! 🙏
